debug: coredump: logging backend should always select LOG

kartben opened the pull request:
debug.coredump.logging_backend tests is using
CONFIG_DEBUG_COREDUMP_BACKEND_LOGGING which depends on LOG.

To reproduce, in 530a3092feef04bc9038f00ac088125b8514aba3, try to run the debug.coredump.logging_backend sample

```
scripts/twister -T tests/subsys/debug/coredump -p m5stack_core2 -j 1
```

The test fails to compile as `DEBUG_COREDUMP_BACKEND` ends up being set to `DEBUG_COREDUMP_BACKEND_INTEL_ADSP_MEM_WINDOW` (**which is likely another bug, as I believe this should be guarded behind e.g. `depends on SOC_FAMILY_INTEL_ADSP`?**), causing a compilation error

```
/home/kartben/zephyrproject/zephyr/subsys/debug/coredump/coredump_backend_intel_adsp_mem_window.c:15:10: fatal error: adsp_memory.h: No such file or directory
   15 | #include <adsp_memory.h>
      |          ^~~~~~~~~~~~~~~
```

kartben committed:
debug: coredump: logging backend should always select LOG

The DEBUG_COREDUMP_BACKEND_LOGGING option should automatically
select LOG Kconfig instead of just depending on it.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

nashif commented on tests/subsys/debug/coredump/prj.conf:
```diff
@@ -1,4 +1,5 @@
 CONFIG_TEST=y
 CONFIG_DEBUG_COREDUMP=y
+CONFIG_LOG=y
```
this fixes just this sample, a more generic fix maybe:

```diff
diff --git a/subsys/debug/coredump/Kconfig b/subsys/debug/coredump/Kconfig
index abeb0d3bda..7031fc6165 100644
--- a/subsys/debug/coredump/Kconfig
+++ b/subsys/debug/coredump/Kconfig
@@ -11,11 +11,11 @@ if DEBUG_COREDUMP

 choice DEBUG_COREDUMP_BACKEND
        prompt "Coredump backend"
-       default DEBUG_COREDUMP_BACKEND_LOGGING if LOG
+       default DEBUG_COREDUMP_BACKEND_LOGGING

 config DEBUG_COREDUMP_BACKEND_LOGGING
        bool "Use Logging subsystem for coredump"
-       depends on LOG
+       select LOG
        help
          Core dump is done via logging subsystem.

```

kartben replied:
good call -- updated.
Thoughts on adding guard(s) for `DEBUG_COREDUMP_BACKEND_INTEL_ADSP_MEM_WINDOW`?

nashif replied:
yeah, we can make this depend on DT_HAS_INTEL_ADSP_MEM_WINDOW_ENABLED

nashif replied:
```
 config DEBUG_COREDUMP_BACKEND_INTEL_ADSP_MEM_WINDOW
        bool "Use memory window for coredump on Intel ADSP"
+       depends on DT_HAS_INTEL_ADSP_MEM_WINDOW_ENABLED
        help
```

kartben replied:
thanks! done.

kartben committed:
debug: coredump: guard selection of ADSP memory window backend

Enabling the core dump via memory window backend only makes sense on
Intel ADSP platform wtih memory windows enabled.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben commented:
@nashif I screwed up the last commit message w.r.t compliance, sorry.
